MesaModel/hamilton/reframe_parameterisation_infrastructure/reframe_tests/parameterisation_tests.py

```diff
@@ -57,3 +64,17 @@ def __init__(self, n_processors, test_id):
             "--model-params",
             params,
         ]
+
+    def extract_mse(self):
```

Comment:
The current output in `run_pipeline.py` was intended as a placeholder - should we modify the script to just output the score rather than the string to save parsing it out here?

Reply:
I think it has to be parsed, because ReFrame only has access to it in the output log. The fact it is also returned by the function doesn't really help us here given the context of a slurm job. Further, if the pipeline only left the score in the output log it would be difficult (impossible?) to differentiate from other numbers that might happen to appear there. Think this way works.

Reply:
OK - yes, it's probably safer to check for the text. Do we need to flag up the case where the text isn't found though? I guess currently it would just not write anything to the file but would it be worth somehow flagging up the failed cases so they can be rerun? Or is it sufficient to do a manual check for missing values after each run of reframe?

Reply:
Currently ReFrame will do a sanity check to see that the string 'Mean squared error: ' is in the log: https://github.com/DurhamARC/classroom-abm/blob/reframe-collect-results/MesaModel/hamilton/reframe_parameterisation_infrastructure/reframe_tests/parameterisation_tests.py#L53

This is evaluated before we even add to the csv. Only tests that pass will append to the csv. If a test fails this is clear to the user in the output from ReFrame and is trackable because no files will be moved into /ddn/data/$USER/reframe_output. Instead the failure is logged in /ddn/data/$USER/reframe_stage.